packages/grafana-data/src/field/fieldComparers.ts

```diff
@@ -79,15 +79,27 @@ const falsyComparer = (a: unknown, b: unknown): number => {
   return 0;
 };
 
-const timestampIndexComparer = (values: number[], reverse: boolean): IndexComparer => {
+const timestampIndexComparer = (values: number[], reverse: boolean, nanos?: number[]): IndexComparer => {
   let mult = reverse ? -1 : 1;
+
+  if (nanos !== undefined) {
+    return (a: number, b: number): number => mult * (values[a] - values[b] || nanos[a] - nanos[b]);
+  }
+
   return (a: number, b: number): number => mult * (values[a] - values[b]);
 };
 
-const timeIndexComparer = (values: unknown[], reverse: boolean): IndexComparer => {
+const timeIndexComparer = (values: unknown[], reverse: boolean, nanos?: number[]): IndexComparer => {
   return (a: number, b: number): number => {
     const vA = values[a];
     const vB = values[b];
+
+    if (nanos !== undefined) {
+      return (
+        (reverse ? timeComparer(vB, vA) : timeComparer(vA, vB)) ||
```

Comment:
let's just rewrite this the same way as the otger function with `let mult = reverse ? -1 : 1`